db.go

```diff
@@ -250,6 +250,164 @@ func newDBMetrics(db *DB, r prometheus.Registerer) *dbMetrics {
 	return m
 }
 
+// DBReadOnly provides APIs for read only operations on a database.
+type DBReadOnly struct {
+	logger  log.Logger
+	dir     string
+	closers []io.Closer
+	mtx     sync.Mutex
+	closed  bool
+}
+
+// OpenDBReadOnly opens DB in the given directory for read only operations.
+func OpenDBReadOnly(dir string, l log.Logger) (*DBReadOnly, error) {
+	if _, err := os.Stat(dir); err != nil {
+		return nil, err
+	}
+
+	if l == nil {
+		l = log.NewNopLogger()
+	}
+
+	db := &DBReadOnly{
+		logger: l,
+		dir:    dir,
+	}
+
+	return db, nil
+}
+
+// Querier loads the wal and returns a new querier over the data partition for the given time range.
+// A goroutine must not handle more than one open Querier.
```

Comment:
What do you mean? It is not thread safe? You can only invoke Querier once from single `DbReadOnly`? Can reword comment for that then?

Reply:
:+1: Not sure either, I think it might mean similar one querier = one goroutine, but no sure 100% (: 
That will work for now.